app/src/main/java/com/github/swent/echo/MainActivity.kt

```diff
@@ -7,22 +7,32 @@ import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.material3.MaterialTheme
 import androidx.compose.material3.Surface
 import androidx.compose.ui.Modifier
+import com.github.swent.echo.authentication.AuthenticationService
 import com.github.swent.echo.compose.navigation.AppNavigationHost
 import com.github.swent.echo.ui.theme.EchoTheme
 import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+import kotlinx.coroutines.runBlocking
 
 @AndroidEntryPoint
 class MainActivity : ComponentActivity() {
+
+    @Inject lateinit var authenticationService: AuthenticationService
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
+        // This happens so quickly, I doesn't make sense to show a loading screen.
+        runBlocking { authenticationService.initialize() }
```

Comment:
This should at least be in the AppNavigationHost, because the authenticationService isn't necessary in the MainActivity

Reply:
The `AppNavigationHost` is a composable, every time it recomposes it would reinitialize the `AuthenticationService`.